c10/cuda/CUDAFunctions.cpp

```diff
@@ -138,6 +139,7 @@ void device_synchronize() {
   if (C10_UNLIKELY(interp)) {
     (*interp)->trace_gpu_device_synchronization(c10::kCUDA);
   }
+  STATIC_SCOPED_WAIT_COUNTER(pytorch.wait_counter.cuda_device_synchronize);
```

Comment:
Does this not require defining a context like the Python-side wait counter?

Reply:
Apparently not, I am guessing it makes an RAII